File: src/Java/CreationalPatterns/AbstractFactory/EraStyles/src/Client/Era.java
package Client;

import Abstraction.EraObject;
import Abstraction.EraObjectStylesFactory;

public class Era {
    private EraObject eraObject;

    public Era(EraObjectStylesFactory eraObjectStylesFactory, char era) {
        switch (era) {
            case 'M':
                eraObject = eraObjectStylesFactory.createMedievalStyleObject();
                break;
            case 'R':
                eraObject = eraObjectStylesFactory.createRenaissanceStyleObject();
                break;
            case 'V':
                eraObject = eraObjectStylesFactory.createVictorianEraStyleObject();
                break;
        }
    }

    public void info() {
        eraObject.showDetails();
    }
}

File: src/PHP/CreationalPatterns/AbstractFactory/EraStyles/Client/Era.php
<?php

namespace CreationalPatterns\AbstractFactory\EraStyles\Client;

use CreationalPatterns\AbstractFactory\EraStyles\Abstraction\EraObject;
use CreationalPatterns\AbstractFactory\EraStyles\Abstraction\EraObjectStylesFactory;

class Era {
    private EraObject $eraObject;

    public function __construct(EraObjectStylesFactory $stylesFactory, $era) {
        switch ($era) {
            case 'M':
                $this->eraObject = $stylesFactory->CreateMedievalStyleObject();
                break;
            case 'R':
                $this->eraObject = $stylesFactory->CreateRenaissanceStyleObject();
                break;
            case 'V':
                $this->eraObject = $stylesFactory->CreateVictorianEraStyleObject();
                break;
        }
    }

    public function info(): void
    {
        $this->eraObject->ShowDetails();
    }
}
